service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java

```diff
@@ -0,0 +1,165 @@
+    var extensionSpy = spy(extensionMap);
+
+    var mockVirtualMachine = mock(VirtualMachine.class);
+    when(mockVirtualMachine.listExtensions()).thenReturn(extensionSpy);
+
+    updateMachineMock = mock(VirtualMachine.Update.class);
+    // verify #apply() is called on removing the extension
+    when(updateMachineMock.withoutExtension(any())).thenReturn(updateMachineMock);
+    when(mockVirtualMachine.update()).thenReturn(updateMachineMock);
+
+    var mockVirtualMachines = mock(VirtualMachines.class);
+    when(mockVirtualMachines.getById(vmId)).thenReturn(mockVirtualMachine);
+    var mockComputeManager = mock(ComputeManager.class);
+    when(mockComputeManager.virtualMachines()).thenReturn(mockVirtualMachines);
+    when(crlService.getComputeManager(eq(azureCloudContext), eq(azureConfig)))
+        .thenReturn(mockComputeManager);
+  }
+
+  @Test
+  void confirmRetry() throws InterruptedException {
+    // this forces the failure to trigger the maybeRetryStatus(...)
+    when(samService.getWsmServiceAccountToken())
+        .thenThrow(new ManagementException("FIRST TIME FAILURE", mockHttpResponse));
+
+    var step =
+        new EnableVmLoggingStep(
+            azureConfig,
+            crlService,
+            resource,
+            landingZoneApiDispatch,
+            samService,
+            mockWorkspaceService);
+    try (MockedStatic mockedUtils = mockStatic(AzureManagementExceptionUtils.class)) {
+      step.doStep(flightContext);
+      mockedUtils.verify(
+          () -> AzureManagementExceptionUtils.maybeRetryStatus(any(ManagementException.class)),
+          times(1));
+    }
+  }
+
+  @Test
+  void confirmBackoutOnFailure() throws InterruptedException {
+
+    when(samService.getWsmServiceAccountToken()).thenReturn(bearerToken.getToken());
+
+    var step =
+        new EnableVmLoggingStep(
+            azureConfig,
+            crlService,
+            resource,
+            landingZoneApiDispatch,
+            samService,
+            mockWorkspaceService) {
+          @Override
+          Optional<ApiAzureLandingZoneDeployedResource> getDataCollectionRuleFromLandingZone() {
+            return Optional.of(mock(ApiAzureLandingZoneDeployedResource.class));
+          }
+
+          @Override
+          void createExtension(FlightContext context, VirtualMachine virtualMachine) {
+            // do nothing ...
+            // the Microsoft classes are package private and not mockable
+          }
+
+          @Override
+          void createDataCollectionRuleAssociation(
+              FlightContext context, ApiAzureLandingZoneDeployedResource dcr, String vmId) {
+            // this has to do with creating the extension ...
+            // and we aren't validating that here
+          }
+        };
+
+    assertEquals(step.doStep(flightContext), StepResult.getStepResultSuccess());
+    verify(updateMachineMock, times(1)).withoutExtension("AzureMonitorLinuxAgent");
```

Comment:
nit: times(1) is the [default](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#verify(T)) and can be omitted.